drivers/uktty/ns16550/ns16550.c

```diff
@@ -29,6 +29,7 @@
  */
 #include <libfdt.h>
 #include <uk/assert.h>
+#include <uk/boot/earlytab.h>
```

Comment:
Does this add ukboot dep?

Reply:
That's a big question as it applies to all libraries initialized via `libukboot`. One option is to wrap this header as well as the registration with `#if CONFIG_LIBUKBOOT` without making the dependency to `libukboot` mandatory in `Config.uk`. On the other hand `libukboot` is so integral that it would be safe to assume that it is always enabled (or implemented as an external library that replaces the native one). @skuenzer thoughts?

Reply:
I think we should add `lib/ukboot` as a dependency as soon as we need headers from it. This is the case when the driver configuration enables early console mode. Otherwise such a `select` should be avoided.